src/pages/form/[id].tsx

```diff
@@ -24,13 +35,71 @@ const Form = () => {
       uiSchema: JSON.parse(data.form.uiSchema),
     };
   }, [data]);
-  if (isLoading) return <DefaultLoadingPage />;
-  if (error) return <ErrorPage errorMessage={error.message} />;
+  const user = useSession();
+
+  const wasSubmitted = useMemo(() => {
+    const wasSubmitted = submssionData?.find(
+      (submission) => submission.submittedById === user?.data?.user?.id,
+    );
+    return wasSubmitted;
+  }, [submssionData, user?.data?.user?.id]);
+  const countdownHome = () => {
+    if (countdown <= 0) {
+      router
+        .push("/")
+        .then()
+        .catch((error) => console.error(error));
+    } else {
+      setCountdown(countdown - 1);
+    }
+  };
+  if ((submitted || wasSubmitted) ?? user.status === "unauthenticated") {
+    setTimeout(countdownHome, 1000);
+  }
   const onSubmit = (data: JSONObject) => {
     submit({ collectionId: router.query.id as string, data });
+    setSubmitted(true);
   };
+
+  if (isLoading || submissionLoading) return <DefaultLoadingPage />;
+  if (error) return <ErrorPage errorMessage={error.message} />;
+  if (submissionError)
+    return <ErrorPage errorMessage={submissionError.message} />;
+  if (user.status === "unauthenticated") {
+    return (
+      <div className="flex min-h-screen flex-col items-center justify-center bg-gradient-to-b from-midnight-sky to-aero">
```

Comment:
We have this same outer div, so return something like

```tsx
<div className="...">
    {body}
</ div>
```

and do a `let body;` and set it in these ifs and have one return

Reply:
You can probably change them all into containers like on line 102 and and have one return